### PR TITLE
Support building Zoekt with Bazel 

### PR DESCRIPTION
**Pull request by jhchabran**
This PR adds support for Building our Zoekt fork with Bazel, which we need in order to build the `cmd/server` container images with Bazel as a few binaries from here are being copied over in the normal build script. 

Having Bazel directly in here will allow to simply reference those needed binaries as external packages and to get them built. I could perhaps have used Gazelle to handle this and generate those on the fly, but Zoekt is rather easy to build, so if felt like a quick win to port do this. 

The only notable fixes I had to do were: 

- to create a `filegroup` for the `/testdata` folder that lives at the root of the repository. 
- add the relevant `gazelle:exclude` to avoid generating buildfiles for `/testdata/repo*`.

I did not patch the CI scripts to use Bazel, it's probably worth doing in another PR instead. 

## Test Plan:

```
bazel build //...
bazel test //... 
```

### CODE REVIEW

File: testdata/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# gazelle:exclude repo
+# gazelle:exclude repo2
```
Comment (jhchabran) [inline testdata/BUILD.bazel:2]: Prevent gazelle from picking up the test Go code.

File: testdata/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+
+filegroup(
+    name = "testdata",
+    srcs = glob(["**"]),
+    visibility = ["//visibility:public"],
+)
```
Comment (jhchabran) [inline testdata/BUILD.bazel:8]: Allows us to use this test data in various go tests requiring it.

File: cmd/zoekt-sourcegraph-indexserver/BUILD.bazel [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        "cleanup_test.go",
+        "index_test.go",
+        "main_test.go",
+        "merge_test.go",
+        "meta_test.go",
+        "owner_test.go",
+        "queue_test.go",
+        "sg_test.go",
+    ],
+    data = glob(["json_schemas/**"]) + ["//testdata:testdata"],
```
Comment (jhchabran) [inline cmd/zoekt-sourcegraph-indexserver/BUILD.bazel:68]: Get both testdata sources.

File: .gitignore [hunk 4]
```diff
@@ -7,3 +7,7 @@ cmd/zoekt-git-index/zoekt-git-index
 .envrc
 .idea
 .direnv
+bazel-bin
+bazel-out
+bazel-testlogs
+bazel-zoekt
```
Comment (jhchabran) [inline .gitignore:13]: Ignore bazel stuff.

**Review by keegancsmith — APPROVED** [review]
LGTM.

How easy will it be for external contributors to use bezel? Or is it fine if we mostly ignore that bazel exists for them, get them to just use go, then as sourcegraphers we run bazel configure for them?

**Comment by jhchabran** [comment]
> How easy will it be for external contributors to use bezel? Or is it fine if we mostly ignore that bazel exists for them, get them to just use go, then as sourcegraphers we run bazel configure for them?

If we enable the CI to build it with bazel, all they would have to do is to run `bazel run :gazelle`, and eventually `bazel run :gazelle-update-repos` if they change the `go.mod`. 

But I'd argue that unless they add new files, there would be zero changes on the buildfiles, so they can safely ignore it. And because I didn't have to do any change clashing with the existing code, it's totally transparent for them (unless if they add stuff). 

I can come back to this and make it build in CI with bazel + update the readme once I'm done with the server image, does that sounds reasonable for you?

**Comment by keegancsmith** [comment]
> > How easy will it be for external contributors to use bezel? Or is it fine if we mostly ignore that bazel exists for them, get them to just use go, then as sourcegraphers we run bazel configure for them?
> 
> If we enable the CI to build it with bazel, all they would have to do is to run `bazel run :gazelle`, and eventually `bazel run :gazelle-update-repos` if they change the `go.mod`.

I think requiring a working bazel is a bit heavy weight given most submissions we get our small in scope. I'm happy as a maintainer to do that stuff for them and let them continue to live in the simple go only work.

> I can come back to this and make it build in CI with bazel + update the readme once I'm done with the server image, does that sounds reasonable for you?

Yes please.

**Pull request merged**